### PR TITLE
[release/6.0] React to dotnet/runtime#57816

### PR DESCRIPTION
**Pull request by dougbu**
- see https://github.com/dotnet/runtime/pull/57816#issuecomment-912755824
- undo much of 18a61fb96c0e

### CODE REVIEW

**Comment by dougbu** [comment]
For now at least, I'm not restoring https://github.com/dotnet/aspnetcore/commit/18a61fb96c0e28e3cc02b81b410279d518eb6734#diff-b7dd9b03307f5fc91be1ae85ffaf732fc154fcd633efa7c597db5e2a3534bd25L92-L101. The CI should confirm whether or not that workaround is needed again.

**Review by wtgodbe — APPROVED** [review]
Nice, LGTM assuming the sharedFx/targeting pack contents are the same

**Comment by dougbu** [comment]
/fyi @ericstj 

/btw @dotnet/aspnet-build, I'm doing this in release/6.0 first because 'main' apparently doesn't have the fix yet.

**Comment by ericstj** [comment]
> 'main' apparently doesn't have the fix yet.

We don't have the S.S.Acls package there, so we need to wait until we can upgrade our packages (eg: Crypto.Xml) to consume stable externals from the 6.0 build

File: src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj [hunk 1]
```diff
@@ -68,14 +68,6 @@ This package is an internal implementation of the .NET Core SDK and is not meant
...
       UndefineProperties="TargetFramework;TargetFrameworks;RuntimeIdentifier;PublishDir" />
 
-    <!-- Omit System.Security.Permissions closure. https://github.com/dotnet/runtime/issues/54341
-         It is brought in by System.Security.Cryptography.Xml for netstandard2.0 compatibility
-         but not required in net6.0 since the types required exist in System.Security.AccessControl. -->
-    <Reference Include="Microsoft.Win32.SystemEvents" ExcludeAssets="All" />
-    <Reference Include="System.Drawing.Common" ExcludeAssets="All" />
-    <Reference Include="System.Security.Permissions" ExcludeAssets="All" />
-    <Reference Include="System.Windows.Extensions" ExcludeAssets="All" />
-
```
Comment (ericstj) [inline src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj:78]: I wonder if you can remove the stuff in `_DisallowedReferenceAssemblies` now?

**Review by ericstj — APPROVED** [review]
LGTM modulo one comment about further cleanup

File: src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj (hunk 1 above)
Comment (dougbu) [reply src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj:78]: Hmm, _maybe_❔ I'll look at a binlog once the CI wraps up.

**Comment by dougbu** [comment]
Looks like we need the change @ericstj mentioned to @ViktorHofer in https://github.com/dotnet/runtime/pull/57816#issuecomment-912791297

The following unexpected assemblies are present in Microsoft.AspNetCore.App.Ref, Microsoft.AspNetCore.App.Runtime, and our various targeting pack and shared framework bundles:
- Microsoft.Win32.SystemEvents.dll
- System.Drawing.Common.dll
- System.Security.Permissions.dll
- System.Windows.Extensions.dll

Will retry the build after dotnet/runtime#58666 is in and reflected in our release/6.0 dependencies

**Comment by ViktorHofer** [comment]
The fix just got merged and should reach you via dependency flow soon.

**Comment by ericstj** [comment]
Looks like this is passing validation builds now.  paging @dougbu

**Comment by dougbu** [comment]
/fyi if I can clean up more of Microsoft.AspNetCore.App.Runtime.csprpoj, I'll file a follow-up issue

**Pull request merged**